doc/book/admin/disaster_recovery.rst

```diff
@@ -1,126 +1,162 @@
+1.  Use ``box.info.election`` to make sure a new master is elected automatically.
 
-      .. code-block:: console
+2.  Remove a crashed master from a replica set.
+
+3.  Set up a replacement for the crashed master on a spare host.
+    Learn more from :ref:`Adding and removing instances <replication-automated-failover-add-remove-instances>`.
+
+See also: :ref:`Testing automated failover <replication-automated-failover-testing>`.
+
+
+.. _admin-disaster_recovery-master_replica_data_loss:
+
+Data loss
+~~~~~~~~~
+
+**Configuration:** master-replica.
+
+**Problem:** Some transactions are missing on a replica after the master has crashed.
+
+**Actions:**
+
+You lose a few transactions in the master
+:ref:`write-ahead log file <index-box_persistence>`, which may have not
+transferred to the replica before the crash. If you were able to salvage the master
+``.xlog`` file, you may be able to recover these.
+
+1.  Find out instance UUID from the crashed master :ref:`xlog <internals-wal>`:
+
+    .. code-block:: console
+
+        $ head -5 var/lib/instance001/*.xlog | grep Instance
+        Instance: 9bb111c2-3ff5-36a7-00f4-2b9a573ea660
+
+2.  On the new master, use the UUID to find the position:
+
+    .. code-block:: tarantoolsession
+
+        app:instance002> box.info.vclock[box.space._cluster.index.uuid:select{'9bb111c2-3ff5-36a7-00f4-2b9a573ea660'}[1][1]]
+        ---
+        - 999
+        ...
+
+3.  :ref:`Play the records <tt-play>` from the crashed ``.xlog`` to the new master, starting from the
+    new master position:
+
+    .. code-block:: console
+
+        $ tt play 127.0.0.1:3302 var/lib/instance001/00000000000000000000.xlog \
+                  --from 1000 \
+                  --replica 1 \
+                  --username admin --password secret
 
-          $ tt play <new_master_uri> <xlog_file> --from 23425 --replica 1
 
 .. _admin-disaster_recovery-master_master:
 
---------------------------------------------------------------------------------
 Master-master
---------------------------------------------------------------------------------
+-------------
+
+**Configuration:** :ref:`master-master <replication-bootstrap-master-master>`.
 
-Configuration: Two masters.
+**Problem:** one master has crashed.
 
-Problem: Master#1 has crashed.
+**Actions:**
 
-Your actions:
+1.  Let the load be handled by another master alone.
 
-1. Let the load be handled by master#2 (effective master) alone.
+2.  Remove a crashed master from a replica set.
+
+3.  Set up a replacement for the crashed master on a spare host.
+    Learn more from :ref:`Adding and removing instances <replication-master-master-add-remove-instances>`.
 
-2. Follow the same steps as in the
-:ref:`master-replica <admin-disaster_recovery-master_replica>` recovery scenario
-to create a new master and salvage lost data.
 
 .. _admin-disaster_recovery-data_loss:
 
---------------------------------------------------------------------------------
 Data loss
```

Comment:
It is confusing to have two 'data loss' sections on different levels.

Reply:
Renamed to `Master-replica/master-master: data loss` for consistency. Still not the best approach but I feel that the entire `Administration` section will be restructured gradually when updating topics to a new config.